transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/Submitter.java

```diff
@@ -30,15 +28,7 @@ static Submitter withExecutor(Executor executor) {
    * @return The submitter.
    */
   static Submitter withDefaultExecutor() {
-    // JDK bug means this warning can't be fixed
-    //noinspection Convert2Diamond
-    return withExecutor(
-        new ThreadPoolExecutor(
-            1,
-            Math.max(1, ForkJoinPool.commonPool().getParallelism()),
-            0L,
-            TimeUnit.MILLISECONDS,
-            new ArrayBlockingQueue<Runnable>(16384)));
+    return ExecutorSubmitter.builder().build();
   }
```

Comment:
@badgerwithagun `ExecutorSubmitter` does not create an executor, but `Submitter` does here. `AutoCloseable` was originally implemented because of this.

Reply:
But you don't need to use this method.

You can just create a `TransactionOutbox` with your own. 

```
var outbox = TransactionOutbox.builder()
   .submitter(ExecutorSubmitter.builder().executor(yourExecutor).build())
   ...
   .builder();
```